File: android/src/main/kotlin/ir/aligator/appmarket/markets/IranApps.kt
package ir.aligator.appmarket.markets

import android.content.Context
import android.content.Intent
import android.net.Uri
import ir.aligator.appmarket.exceptions.FeatureNotSupportedException


class IranApps(private val context: Context) : AppMarket {
    override val id: Int = 2
    override val name: String = "iranApps"

    override fun showAppPage(packageId: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.setPackage("iranapps://app/$packageId")
        intent.data = Uri.parse("")
        context.startActivity(intent)
    }

    override fun showAppComments(packageId: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.setPackage("ir.tgbs.android.iranapp")
        intent.data = Uri.parse("iranapps://app/$packageId?a=comment&r=5")
        context.startActivity(intent)
    }

    override fun showDeveloperApps(devId: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.setPackage("ir.tgbs.android.iranapp")
        intent.data = Uri.parse("iranapps://user/$devId")
        context.startActivity(intent)
    }

    override fun getVersionCode(packageId: String): Long {
        throw FeatureNotSupportedException(name, "getVersionCode")
    }
}

File: lib/src/iranapps.dart
import 'package:android_intent/android_intent.dart';
import 'package:iran_appmarket/iran_appmarket.dart';

class IranApps implements IAppMarket {
  static const iranapps = 'ir.tgbs.android.iranapp';

  @override
  void showAppPage(String packageName) {
    AndroidIntent(
      action: "android.intent.action.VIEW",
      data: 'iranapps://app/$packageName',
      arguments: {
        "android.intent.extra.PACKAGE_NAME" : iranapps,
      },
    ).launch();
  }

  @override
  void showAppComments(String packageName) {
    AndroidIntent(
      action: "android.intent.action.VIEW",
      data: 'iranapps://app/$packageName?a=comment&r=5',
      arguments: {
        "android.intent.extra.PACKAGE_NAME" : iranapps,
      },
    ).launch();
  }

  @override
  void showDeveloperApps(String developerId) {
    AndroidIntent(
      action: "android.intent.action.VIEW",
      data: 'iranapps://user/$developerId',
      arguments: {
        "android.intent.extra.PACKAGE_NAME" : iranapps,
      },
    ).launch();
  }
}